pkg/controller/account/ec2.go

```diff
@@ -0,0 +1,196 @@
+package account
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/go-logr/logr"
+	"github.com/openshift/aws-account-operator/pkg/awsclient"
+)
+
+// InitializeSupportedRegions concurrently calls InitalizeRegion to create instances in all supported regions
+// This should ensure we don't see any AWS API "PendingVerification" errors when launching instances
+func (r *ReconcileAccount) InitializeSupportedRegions(reqLogger logr.Logger, regions map[string]map[string]string, creds *sts.AssumeRoleOutput) error {
+	// Create some channels to listen and error on when creating EC2 instances in all supported regions
+	ec2Notifications, ec2Errors := make(chan string), make(chan string)
+
+	// Make sure we close our channels when we're done
+	defer close(ec2Notifications)
+	defer close(ec2Errors)
+
+	for region := range regions {
+		go r.InitializeRegion(reqLogger, region, regions[region]["initializationAMI"], ec2Notifications, ec2Errors, creds)
+	}
+
+	// Wait for all the regions to complete
+	for i := 0; i < len(regions); i++ {
+		select {
+		case msg := <-ec2Notifications:
+			reqLogger.Info(msg)
+		case errMsg := <-ec2Errors:
+			reqLogger.Error(errors.New(errMsg), errMsg)
+		}
+	}
+
+	reqLogger.Info("Completed initializing all supported regions")
+
+	return nil
+}
+
+// InitializeRegion sets up a connection to the AWS `region` and then creates and terminates an EC2 instance
+func (r *ReconcileAccount) InitializeRegion(reqLogger logr.Logger, region string, ami string, ec2Notifications chan string, ec2Errors chan string, creds *sts.AssumeRoleOutput) error {
```

Comment:
Would it make sense to wrap some of the items passed to this function into a struct? A lot of the same args are passed to `BuildandDestroyEC2Instances`

Reply:
Agree but I think we can refactor this after

Reply:
:+1;